### ISSUE TITLE
Insufficient documentation for getExpansion_2()

### ISSUE THREAD

**Issue by okenk** [pfmc-assessments/PacFIN.Utilities#136]
**Describe the bug**
I am unclear what the structure of the catch data frame should be that I provide in getExpansion_2(). Should it be raw fish ticket data? (It looks like not?) Data that has been aggregated by fleet and year? What should the columns be called? The vignettes describe the general approach to expansions, but neither the vignette nor the help file for the function show working examples that would help users understand how to format data frames.

**Expected behavior**
User should be able to read help files and understand what inputs the function expects.

**Comment by iantaylor-NOAA** [comment]
The function should definitely be better documented and I think a detailed vignette showing workflow for processing BDS data would be helpful.

In general, the functions for working with BDS data only add columns and do not modify the existing columns, where the same original data extraction can be passed to each. You could pipe them all together (although in the past folks have more often saved the intermediate steps as separate objects):
```
Pdata_expanded <- Pdata |> 
  cleanPacFIN() |> 
  getExpansion_1() |> 
  getExpansion_2()
```
In order to use the resulting expansion values, you need to add a calculation like 
` Pdata_expanded $Final_Sample_Size <- capValues(Pdata_expanded$Expansion_Factor_1_L * Pdata_expanded$Expansion_Factor_2)`

Here's the code I used for petrale in 2023 https://github.com/pfmc-assessments/petrale/blob/main/Rscripts/process_pacfin_bds.R
which is longer than necessary because we were exploring multiple model structure assumptions. I think that script may have been based on this Dover Sole script from @chantelwetzel-noaa: https://github.com/chantelwetzel-noaa/dover_sole_2021/blob/master/code/commercial_comps.R.

**Comment by chantelwetzel-noaa** [comment]
I think documentation can always be improved.  If you have specific thoughts on what would have been useful for you to know in the documentation, I suggest creating a branch and making the corresponding edits. 

@kellijohnson-NOAA has created an example vignette for this package that is available in the rm-doSexRatio branch.  There are a couple of minor changes needed in this branch and we will then get it merged into main which will make it more visible for users.

**Comment by iantaylor-NOAA** [comment]
Thanks @chantelwetzel-noaa, I knew I'd seen a more recent example somewhere.
Direct link for now (until it gets merged) is https://github.com/pfmc-assessments/PacFIN.Utilities/blob/rm-doSexRatio/vignettes/sablefish_example.Rmd.

**Comment by okenk** [comment]
Thanks, the vignette is a lot more helpful than scrounging through old code from past assessments. It would be nice to prioritize making that more accessible to users. I think my questions in the issue summarize what specific details would have helped, but if I had found the vignette I would have been fine.

This goes beyond the documentation, and is definitely a lot more work and not an immediate issue, but did slow me down as it was unexpected: I am curious how the format for the catch data frame was developed. I avoid wide data frames that seem to be required by the function because 1) stock synthesis uses long data formats, 2) ggplot uses long data formats, and 3) it is a lot easier to manipulate data in a long format using dplyr. The limited use for wide data structures seems to be in assessment documents (and other similar communications), and also this function.

Finally, I did not know that there were {PacFIN.Utilities} functions to summarize catches. I had already done it myself with 2 lines of code, so did not even think to look for a function. In terms of documentation, it may be helpful to avoid assuming catches were worked up using a particular function that results in a particular data structure (that you have to reformat anyway if you want to use ggplot).

**Comment by iantaylor-NOAA** [comment]
@okenk, these are good ideas.
I think Andi Stephens started work on the initial PacFIN.Utilities package in 2012, based on one-off examples that others had developed for years prior to that, when the team didn't really have any understanding of good coding practices as we know them now and much of the tidyverse didn't yet exist.

The package could use a full rewrite to align better with best coding practices, but given everyone's workloads, that's a long way off. In the interim, if there are some low hanging fruit for easy fixes or improvements in the documentation in addition to the changes that @chantelwetzel-noaa is working on in #135, pull requests and are welcome.

**Comment by okenk** [comment]
I fully agree reformatting data frames would require a more wholesale overhaul that no one is going to do in the middle of an assessment cycle! Was not trying to propose any immediate changes on that front. :)

**Comment by iantaylor-NOAA** [comment]
The vignette is now live on the website for this package https://pfmc-assessments.github.io/PacFIN.Utilities/articles/sablefish_example.html including example use of the new [get_pacfin_expansions()](https://pfmc-assessments.github.io/PacFIN.Utilities/reference/get_pacfin_expansions.html) function. The parts of this discussion related to formatting the input catch are now in the separate issue #138 thanks to @okenk, so I think we can close this now.
